www/js/survey/enketo/enketo-add-note-button.js

```diff
@@ -0,0 +1,287 @@
+
+  $scope.displayLabel = () => {
+    const localeCode = $translate.use();
+    // if already filled in
+    //   return $scope.notesConfig?.['filled-in-label']?.[localeCode];
+    return $scope.notesConfig?.['not-filled-in-label']?.[localeCode];
+  }
+
+  $scope.getPartialTimeUseResponse = () => {
+    const timebounds = $scope.timeBounds(); // for trips, these times are start and end of trip
+                                            // for places, it is enter and exit time (end of one trip and start of next)
+
+    // Prefill the time bounds of the trip into the survey
+    const startDayAndTime = timebounds.start_fmt_time.split('T');
+    const startHMS = startDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const startTimezoneCode = startDayAndTime[1].split('-')[1];
+    const startTime = `${startHMS}.000-${startTimezoneCode}`
+    const endDayAndTime = timebounds.end_fmt_time.split('T');
+    const endHMS = endDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const endTimezoneCode = endDayAndTime[1].split('-')[1];
+    const endTime = `${endHMS}.000-${endTimezoneCode}`
+
+    // TODO: Hardcoding a partial TimeUseSurvey response for now
+    // Can we come up with a more generic and more elegant way to do this ?
+    return {
+      data: {
+        name: "TimeUseSurvey",
+        xmlResponse:
+        `<a88RxBtE3jwSar3cwiZTdn xmlns:jr=\"http://openrosa.org/javarosa\" xmlns:orx=\"http://openrosa.org/xforms\" id=\"a88RxBtE3jwSar3cwiZTdn\">
+          <start>${timebounds.start_fmt_time}</start>
+          <end>${timebounds.end_fmt_time}</end>
+          <group_hg4zz25>
+            <Date>${startDayAndTime[0]}</Date>    ${/* YY:MM:DD */''}
+            <Start_time>${startTime}</Start_time> ${/* HH:MM:SS.mmm-HH:MM */''}
+            <End_time>${endTime}</End_time>
+          </group_hg4zz25>
+        </a88RxBtE3jwSar3cwiZTdn>`
+      }
+    };
+  }
+
+  const getScrollElement = function() {
+    if (!$scope.scrollElement) {
+        console.log("scrollElement is not cached, trying to read it ");
+        const ionItemElement = $element.closest('ion-item')
+        if (ionItemElement) {
+            console.log("ionItemElement is defined, we are in a list, finding the parent scroll");
+            $scope.scrollElement = ionItemElement.closest('ion-content');
+        } else {
+            console.log("ionItemElement is defined, we are in a detail screen, ignoring");
+        }
+    }
+    // TODO: comment this out after testing to avoid log spew
+    console.log("Returning scrollElement ", $scope.scrollElement);
+    return $scope.scrollElement;
+  }
+
+  $scope.openPopover = function ($event, trip, inputType) {
+    const surveyName = $scope.notesConfig.surveyName;
+    console.log('About to launch survey ', surveyName);
+
+    let partialTimeUseResponse;
+    if (surveyName == 'TimeUseSurvey') {
+      partialTimeUseResponse = $scope.getPartialTimeUseResponse();
+    }
+
```

Comment:
This needs to be more more generic, right? I thought we could specify any type of survey for the notes button.